### ISSUE TITLE
Uncaught ReferenceError: global is not defined when using Vite (+Vue)

### ISSUE THREAD

**Issue by Uninen** [realm/realm-js#3866]
## Goals

Initialize Realm in Vue app using Realm Web SDK and Vite.

## Expected Results

App initializes without errors.

## Actual Results

App crashes with error: `Uncaught ReferenceError: global is not defined `

(Possibly related: https://github.com/vitejs/vite/issues/2618 )

## Steps to Reproduce

1. Create an empty Vite app
2. Add minimal code to initialize Realm

The error goes away when I shim `global` with this in my main template: 
```html
<script>window.global = window;</script>
```

## Code Sample

```ts
import * as Realm from 'realm-web'

const app: Realm.App = new Realm.App({ id: 'my-app-id' })
const credentials = Realm.Credentials.anonymous()
try {
  const user = await app.logIn(credentials)
  console.log('user: ', user)
} catch (err) {
  console.error('Failed to log in', err)
}
```

## Version of Realm and Tooling

- Realm JS SDK Version: 1.3.0
- Node: v14.15.1
- Vite: v2.4.2
- Vue: 3.1

**Comment by takameyer** [comment]
@Uninen It appears that vitejs/vite#2618 defines the problem quite well. We will investigate if it's feasible to avoid using the global keyword in `realm-web`. In the meantime, it is good you have a workaround :)

**Comment by ClearedFram3** [comment]
I am having a similar issue in sveltekit. Everything breaks because `global is undefined` unless I dynamically import realm-web after calling this 
```
Object.defineProperty(globalThis, 'global', {
	value: globalThis,
	enumerable: true,
	configurable: true,
	writable: true
});
```

**Comment by kraenhansen** [comment]
@Uninen would you mind sharing the entire stacktrace when the error occurs?

**Comment by Uninen** [comment]
Sure. Not sure how useful this is though so I made a quick repro project that is very easy to set up (just clone and yarn install).

Here's the stacktrace:
```
05:54:21.520 [vite] connecting... client:180:9
05:54:21.552 [vite] connected. client:202:21
05:54:21.782 Uncaught ReferenceError: global is not defined
    <anonymous> http://localhost:3000/node_modules/.vite/realm-web.js?v=e0afb5d8:1676
realm-web.js:1676:19
    <anonymous> http://localhost:3000/node_modules/.vite/realm-web.js?v=e0afb5d8:1676
    InnerModuleEvaluation self-hosted:2384
    InnerModuleEvaluation self-hosted:2384
    InnerModuleEvaluation self-hosted:2384
    evaluation self-hosted:2335
```

and here's a minimal repro project (with detailed setup instructions in the top of the README): https://github.com/Uninen/realm-js-bug-3866

**Comment by ibilux** [comment]
> I am having a similar issue in sveltekit. Everything breaks because `ReferenceError: global is not defined`
> The current fix is to dynamically import `realm-web` after calling this
> 
> ```
> Object.defineProperty(globalThis, 'global', {
> 	value: globalThis,
> 	enumerable: true,
> 	configurable: true,
> 	writable: true
> });
> ```

Can you explain more how did you fix this ?

**Comment by igorovic** [comment]
Same issue with an angular project. 

here is the stack trace 
```
Uncaught ReferenceError: global is not defined
    at Module.5953 (bson.browser.esm.js:2044)
    at __webpack_require__ (bootstrap:19)
    at Module.7053 (index.js:4)
    at __webpack_require__ (bootstrap:19)
    at Module.5041 ($_lazy_route_resources|lazy|groupOptions: {}|namespace object:13)
    at __webpack_require__ (bootstrap:19)
    at Module.6747 (app.component.html:1)
    at __webpack_require__ (bootstrap:19)
    at Module.4431 (environment.ts:16)
    at __webpack_require__ (bootstrap:19)
```

and a repo to reproduce the issue: [repo](https://github.com/igorovic/realm-js-issue-3866)

**Comment by kraenhansen** [comment]
With @igorovic's stacktrace I can see that the issue might be rooted in the `bson` dependency that Realm Web has a dependency on. We recently upgraded this dependency and I believe this reference to `global` should no longer be there.

Can you please verify that this has been fixed by upgrading to Realm Web v1.5.1? Please reopen the issue if that's not the case.
